logstash-core/src/main/java/org/logstash/launchers/JvmOptionsParser.java

```diff
@@ -112,17 +108,28 @@ public static void main(final String[] args) throws InterruptedException, IOExce
                 .findFirst();
     }
 
-    private void parse(Path jvmOptionsFile) throws IOException, JvmOptionsFileParserException {
-        final List<String> jvmOptionsContent = parseJvmOptions(jvmOptionsFile);
+    private void parseAndInjectEnvironment(Optional<Path> jvmOptionsFile) throws IOException, JvmOptionsFileParserException {
+        final List<String> jvmOptionsContent = new ArrayList<>();
+
+        if (jvmOptionsFile.isPresent()) {
+            System.err.format("Processing jvm.options file at `%s`\n", jvmOptionsFile.get().toString());
+            jvmOptionsContent.addAll(parseJvmOptions(jvmOptionsFile.get()));
+        } else {
+            System.err.println("Warning: no jvm.options file found.");
+        }
+
         final String lsJavaOpts = System.getenv("LS_JAVA_OPTS");
         if (lsJavaOpts != null && !lsJavaOpts.isEmpty()) {
+            System.err.println("Appending jvm options from environment LS_JAVA_OPTS");
             jvmOptionsContent.add(lsJavaOpts);
         }
+
         System.out.println(String.join(" ", jvmOptionsContent));
     }
 
     private List<String> parseJvmOptions(Path jvmOptionsFile) throws IOException, JvmOptionsFileParserException {
         if (!jvmOptionsFile.toFile().exists()) {
+            System.err.format("Warning: jvm.options file does not exist or is not readable: `%s`\n", jvmOptionsFile);
             return Collections.emptyList();
         }
```

Comment:
I believe this block becomes redundant because of https://github.com/elastic/logstash/pull/13525/files#diff-8b7e8ef187a3b168614dff225f36286922368b92a2c4352d0b029746df7e308dR114. If the file is not there or can't be read then parseJvmOptions will never be called

Reply:
Agree and on the other path a warn message is already printed into stderr

Reply:
We could also remove the parameter check, because this method is always called from a method that already checks for `jvmOptionsFile.isPresent()` . But keeping the guard saves us from arguments errors, especially if this code is moved around and we miss to check the param.

Reply:
I understand the concerrn with moving around the code, so we can remove the check from the top, and have only 
```java
jvmOptionsContent.addAll(parseJvmOptions(jvmOptionsFile.get()));
```

And `parseJvmOptions` will take care of detecting if the file exists and returning an empty array in that case?

Reply:
yep, I keep the warn to stdout in case the jvmOptionsFile path doesn't exists

Reply:
I think that we have to keep both `if`s, because:
`if (jvmOptionsFile.isPresent()) {` checks if the `Optional` jvmOptionsFile contains any data.
The other inner if `if (!jvmOptionsFile.toFile().exists()) {` checks if the `Optional` content is effectively an existing file

Reply:
so we could have the condition where the Optional.isPresent() is true but the file path contained into doesn't exists. Pretty weird condition

Reply:
I've moved all the checks of Optional presence and it's content (file path) existence into the `parseJvmOptions` method, to have all in the same place